src/components/LayerSwitcher/AddLayerButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Box } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import { Layer, useMapStateContext } from '../utils/MapStateContext';
+import { t } from '../../services/intl';
+import { AddCustomDialog } from './AddCustomLayer';
+
+export const AddUserLayerButton = ({ setUserLayers }) => {
+  const { setActiveLayers } = useMapStateContext();
+  const [isOpen, setOpen] = React.useState(false);
+
+  return (
+    <>
+      <Box m={2} mt={6}>
+        <Button size="small" color="secondary" onClick={() => setOpen(true)}>
+          {t('layerswitcher.add_layer_button')}
+        </Button>
+      </Box>
+
+      <AddCustomDialog
+        onClose={() => setOpen(false)}
+        isOpen={isOpen}
+        save={(layer) => {
+          const { url, name, max_zoom: maxzoom, attribution, bbox } = layer;
+
+          const newLayer: Layer = {
+            type: 'user',
+            maxzoom,
+            name,
+            url,
+            attribution,
```

Comment:
Can we perhaps leave `attribution` as an array? I think it may help the code get simpler.